linter/src/crow/crow.cpp

```diff
@@ -43,83 +44,44 @@ get_server_thread()
             std::string uid = req.url_params.get("uid");
             std::string algo_id = req.url_params.get("algo_id");
 
-            // Watchdog to check for crash/pending after
-            // LINT_ABSOLUTE_TIMEOUT_SECONDS seconds (default 130s)
-            std::thread check_pending_thread([algo_id, uid]() {
-                for (int i = 0; i <= LINT_ABSOLUTE_TIMEOUT_SECONDS; i++) {
-                    if (nutc::client::get_algo_status(uid, algo_id)
-                        != nutc::client::LintingResultOption::PENDING) {
-                        log_i(
-                            crow_watchdog,
-                            "Algo id {} for uid {} firebase result detected as set by "
-                            "crow-level watchdog {}s",
-                            algo_id,
-                            uid,
-                            i
-                        );
-                        return;
-                    }
-                    std::this_thread::sleep_for(std::chrono::seconds(1));
-                }
-                // Check status from Firebase
-                nutc::client::LintingResultOption linting_status =
-                    nutc::client::get_algo_status(uid, algo_id);
+            std::optional<std::string> algo_code = nutc::client::get_algo(uid, algo_id);
+            if (!algo_code.has_value()) {
+                nutc::client::set_lint_failure(
+                    uid,
+                    algo_id,
+                    fmt::format(
+                        "[linter] FAILURE - could not find algo {} for id {}\n",
+                        algo_id,
+                        uid
+                    )
+                );
+                crow::json::wvalue response = crow::json::wvalue({
+                    {"linting_status",
+                     static_cast<int>(client::LintingResultOption::UNKNOWN)}
+                });
+
+                res.body = response.dump();
+                res.code = 200;
```

Comment:
wouldn't this be 404 or 400?

Reply:
Meant to leave a TODO about this, currently 200 is expected by the rust server so I am just preserving past behavior.